packages/test/test-end-to-end-tests/src/test/SummarizeFetchValidation.spec.ts

```diff
@@ -0,0 +1,314 @@
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        summarizer1.close();
+    });
+
+    it("Summarizing consecutive times should not fetch", async () => {
+        const summarizer1 = await createSummarizer(provider, mainContainer);
+
+        let versionWrap = await incrementCellValueAndRunSummary(summarizer1, 1 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+
+        versionWrap = await incrementCellValueAndRunSummary(summarizer1, 2 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+
+         // Create 5 data stores and add their handles to mark it as referenced.
+        const dataStore2 = await dataStoreFactory1.createInstance(mainDataStore._context.containerRuntime);
+        const dataStore3 = await dataStoreFactory1.createInstance(mainDataStore._context.containerRuntime);
+        const dataStore4 = await dataStoreFactory1.createInstance(mainDataStore._context.containerRuntime);
+        const dataStore5 = await dataStoreFactory1.createInstance(mainDataStore._context.containerRuntime);
+        const dataStore6 = await dataStoreFactory1.createInstance(mainDataStore._context.containerRuntime);
+
+        mainDataStore._root.set("dataStore2", dataStore2.handle);
+        mainDataStore._root.set("dataStore3", dataStore3.handle);
+        mainDataStore._root.set("dataStore4", dataStore4.handle);
+        mainDataStore._root.set("dataStore5", dataStore5.handle);
+        mainDataStore._root.set("dataStore6", dataStore6.handle);
+
+        versionWrap = await incrementCellValueAndRunSummary(summarizer1, 3 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        await provider.ensureSynchronized();
+        summarizer1.close();
+    });
+
+    it("Second summarizer from latest should not fetch", async () => {
+        const summarizer1 = await createSummarizer(provider, mainContainer);
+
+        let versionWrap = await incrementCellValueAndRunSummary(summarizer1, 1 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        versionWrap = await incrementCellValueAndRunSummary(summarizer1, 2 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        await provider.ensureSynchronized();
+        summarizer1.close();
+
+        const value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0, "1");
+        assert(value === 3, "Value matches expected");
+
+        const summarizer2 = await createSummarizer(provider, mainContainer);
+        versionWrap = await incrementCellValueAndRunSummary(summarizer2, 4 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "There should be one fetch after second summary");
```

Comment:
This message says there should be one fetch but the assert is that fetchCount should be 0?